src/backend/cuda/platform.hpp

```diff
@@ -116,9 +159,29 @@ class DeviceManager
 
         int setActiveDevice(int device, int native = -1);
 
+        void resetcublasHandle(int device);
+
+        void resetcusolverHandle(int device);
+
+        void resetcusparseHandle(int device);
+
         int activeDev;
         int nDevices;
         cudaStream_t streams[MAX_DEVICES];
+
+        std::unique_ptr<MemoryManager> memManager;
+
+        std::unique_ptr<MemoryManagerPinned> pinnedMemManager;
+
+        std::unique_ptr<InteropManager> gfxManager;
+
+        cufft::cuFFTPlanner cufftManagers[MAX_DEVICES];
+
+        std::unique_ptr<cublas::cublasHandle> cublasHandles[MAX_DEVICES];
+
+        std::unique_ptr<cusolver::cusolverDnHandle> cusolverHandles[MAX_DEVICES];
+
+        std::unique_ptr<cusparse::cusparseHandle> cusparseHandles[MAX_DEVICES];
```

Comment:
In some cases, the respective resource manager class itself uses some information that is initialised during the creation of DeviceManager class which is causing infinite loop. Therefore, for such cases, i am lazying initialising them using unique_ptrs.